src/int_id.h

```diff
@@ -48,6 +48,11 @@ class int_id
          */
         int_id( const string_id<T> &id );
 
+        /**
+         * constexpr constructor, use to allow compilers to optimize some checks to compile time.
+         */
+        explicit constexpr int_id( const char *id ) : int_id( string_id<T>( id ) ) {}
```

Comment:
I'm not sure that there is any sense to have `int_id` as a constexpr. Currently the string_id -> int_id conversion makes sense only in runtime, when the map of available string_ids is populated. 
In fact, we should have some kind of static check that prohibits creation of int ids with static lifetime.

Reply:
The intention was to push the compiler into evaluation the string_id in compile time when possible, but since transferring to templates, this is no longer needed, my bad